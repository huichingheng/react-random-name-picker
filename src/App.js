import React, { Component } from "react";
import math from "mathjs";
import RandomButton from "./RandomButton";
import List from "./List";
import InputForm from "./InputForm";

class App extends Component {
  constructor() {
    super();
    this.state = {
      names: ["gordon", "sahil", "david", "sally", "jane", "alice"],
      luckyWinnerIndex: null,
      value: ""
    };
  }

  render() {
    return (
      <div>
        <h1 id="app-title">Hello, random name picker</h1>
        {/* <Input handleChange={event => this.handleChange(event)} /> */}
        <InputForm
          handleOnChange={event => this.handleChange(event)}
          handleClickAdd={() => this.handleClickAdd()}
        />
        <List
          names={this.state.names}
          luckyWinnerIndex={this.state.luckyWinnerIndex}
        />
        <RandomButton handleClick={() => this.handleClick()} />
      </div>
    );
  }

  handleClick() {
    const number = math.randomInt(this.state.names.length);
    console.log(number);
    this.setState({ luckyWinnerIndex: number });
  }

  handleClickAdd() {
    this.setState({ names: [...this.state.names, this.state.value] });
  }

  handleChange(event) {
    this.setState({ value: event.target.value });
  }
}

export default App;
