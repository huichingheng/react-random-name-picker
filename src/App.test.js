import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import ShallowRender from "react-test-renderer/shallow";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders div with some text", () => {
  const renderer = new ShallowRender();
  renderer.render(<App />);
  const output = renderer.getRenderOutput();

  expect(output).toMatchSnapshot();
});
