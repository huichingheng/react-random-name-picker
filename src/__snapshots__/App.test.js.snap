// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders div with some text 1`] = `
<div>
  <h1
    id="app-title"
  >
    Hello, random name picker
  </h1>
  <InputForm
    handleClickAdd={[Function]}
    handleOnChange={[Function]}
  />
  <List
    luckyWinnerIndex={null}
    names={
      Array [
        "gordon",
        "sahil",
        "david",
        "sally",
        "jane",
        "alice",
      ]
    }
  />
  <RandomButton
    handleClick={[Function]}
  />
</div>
`;
